File: 402_2016_1/class_22/SavingWithNSCoding/SavingWithNSCoding/Person.m
//
//  Person.m
//  SavingWithNSCoding
//
//  Created by CICCC1 on 2016-07-19.
//  Copyright © 2016 Ideia do Luiz. All rights reserved.
//

#import "Person.h"

@implementation Person

- (void)encodeWithCoder:(NSCoder *)aCoder
{
    [aCoder encodeObject:self.name forKey:@"name"];
    [aCoder encodeObject:self.ID forKey:@"ID"];
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder
{
    NSString *name = [aDecoder decodeObjectForKey:@"name"];
    NSString *ID = [aDecoder decodeObjectForKey:@"ID"];
    
    self = [[Person alloc] initWithID:ID name:name];
    return self;
}

- (instancetype)initWithID:(NSString *)ID name:(NSString *)name
{
    self = [super init];
    if (self)
    {
        _name = name;
        _ID = ID;
    }
    
    return self;
}

@end

File: 402_2016_2/SavingWithNSCoding/SavingWithNSCoding/Person.swift
//
//  Person.swift
//  SavingWithNSCoding
//
//  Created by Luiz on 2016-11-16.
//  Copyright © 2016 Ideia do Luiz. All rights reserved.
//

import UIKit

class Person: NSObject, NSCoding
{
    var name:NSString?
    var age:NSNumber?
    
    init(name: NSString?, age: NSNumber?) {
        super.init()
        
        self.age = age
        self.name = name
    }
    
    func encode(with aCoder: NSCoder) {
        aCoder.encode(self.name, forKey: "nameKey")
        aCoder.encode(self.age, forKey: "ageKey")
    }
    
    convenience required init?(coder aDecoder: NSCoder) {
        let name = aDecoder.decodeObject(forKey: "nameKey") as? NSString
        let age = aDecoder.decodeObject(forKey: "ageKey") as? NSNumber
        
        self.init(name: name, age: age)
    }
}
